src/SearchableEntityInterface.php

```diff
@@ -7,7 +7,13 @@
 {
     public function getIndexName();
 
+    /**
+     * @return array Serialized entity fields
+     */
     public function getSearchableArray();
 
+    /**
+     * @return string|int|array Serialized entity identifier
```

Comment:
The return value of `getId` can be a string? When?

Reply:
@nunomaduro I think you are correct, since we are normalizing and not serializing here. I think Symfony Serializer can normalize value to string, but I will have to check this.

Logically however I think `getId()` is meant to return string (based on the original code), since it lands in `objectId`. Maybe this actually should **always** return scalar?

Reply:
Yes, arrays will break. The engine will convert the objectID to a string, so any scalar would do.